api/clients/v2/validator/retrieval_worker.go

```diff
@@ -0,0 +1,709 @@
+	case <-w.ctx.Done():
+		return nil, fmt.Errorf("retrieval worker context cancelled: %w", w.ctx.Err())
+	case decodeResponse := <-w.decodeResponseChan:
+		if decodeResponse.Err == nil {
+			return decodeResponse.Blob, nil
+		} else {
+			return nil, fmt.Errorf("failed to decode blob: %w", decodeResponse.Err)
+		}
+	}
+}
+
+// downloadChunks downloads the chunk data from the specified operator.
+func (w *retrievalWorker) downloadChunks(operatorID core.OperatorID) {
+	if w.detailedLogging {
+		w.logger.Debug("downloading chunks",
+			"operator", operatorID.Hex(),
+			"blobKey", w.blobKey.Hex())
+	}
+
+	// Report back to the control loop when the download started. This may be later than when
+	// the download was scheduled if there is contention for the connection pool.
+	w.downloadStartedChan <- &downloadStarted{
+		operatorID:    operatorID,
+		downloadStart: w.timeSource(),
+	}
+
+	// Unless this has been overridden for testing, this is just a call to standardDownloadChunks().
+	reply, err := w.downloadChunksFunction(w.ctx, w.blobKey, operatorID)
+
+	w.downloadCompletedChan <- &validatorDownloadCompleted{
+		OperatorID: operatorID,
+		Reply:      reply,
+		Err:        err,
+	}
+}
+
+// downloadChunksFunction is the default function used to download chunks from a validator node. This may not
+// be called if
+func (w *retrievalWorker) standardDownloadChunks(
+	_ context.Context,
+	_ v2.BlobKey,
+	operatorID core.OperatorID,
+) (*grpcnode.GetChunksReply, error) {
+
+	ctx, cancel := context.WithTimeout(w.downloadAndVerifyContext, w.config.DownloadTimeout)
+	defer cancel()
+
+	// TODO we can get a tighter bound?
+	maxBlobSize := 16 * units.MiB // maximum size of the original blob
+	encodingRate := 8             // worst case scenario if one validator has 100% stake
+	fudgeFactor := units.MiB      // to allow for some overhead from things like protobuf encoding
+	maxMessageSize := maxBlobSize*encodingRate + fudgeFactor
+
+	opInfo := w.operatorInfo[operatorID]
+	conn, err := grpc.NewClient(
+		opInfo.Socket.GetV2RetrievalSocket(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMessageSize)),
+	)
+	defer func() {
+		err := conn.Close()
+		if err != nil {
+			w.logger.Error("validator retriever failed to close connection", "err", err)
+		}
+	}()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create connection to operator %s: %w", operatorID.Hex(), err)
+	}
+
+	n := grpcnode.NewRetrievalClient(conn)
+	request := &grpcnode.GetChunksRequest{
+		BlobKey:  w.blobKey[:],
+		QuorumId: uint32(w.quorumID),
+	}
+
+	reply, err := n.GetChunks(ctx, request)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chunks from operator %s: %w", operatorID.Hex(), err)
+	}
+
+	return reply, nil
+}
+
+// deserializeAndVerifyChunks deserializes the chunks from the GetChunksReply and sends them to the chunksChan.
+func (w *retrievalWorker) deserializeAndVerifyChunks(
+	_ context.Context,
```

Comment:
I think this context parameter can be removed

Reply:
Not actually fixed, unused context is still there

Reply:
Oops... actually removed now.